### ISSUE TITLE
[DataTable] Single row selection

### ISSUE THREAD

**Issue by mstn** [tleunen/react-mdl#383]
I was wondering if it is possible to select only a single row in a data table or to attach click events to single rows.

**Comment by tleunen** [comment]
I agree, I implemented it in our own library at work. But I'm not sure about it for MDL, I don't see it in the initial spec :/ (See https://material.google.com/components/data-tables.html#data-tables-specs)

Maybe you can ask the MDL team? https://github.com/google/material-design-lite
It might just be something missing in the documentation. 

But yes this can be done quite easily in react-mdl. I just don't want to do too many things that are not in the MD spec, or the MDL library.

**Comment by Augustin82** [comment]
Something I don't understand: if I add a 'className' attribute to each row of the rows object I pass to DataTable, the classes are added to the tr. However, it does not work that way with the 'style' attribute. This is obviously a problem when defining all your styles inline, instead of via external CSS.

I have added it myself as a quick'n'dirty hack, but I cannot use it if it's not in the official repo...

The change is on line 138 of `react-mdl/lib/DataTable/Table.js` : 
```diff
-- { key: row[rowKeyColumn] || row.key || idx, className: row.className },
++ { key: row[rowKeyColumn] || row.key || idx, className: row.className, style: row.style },
```

**Comment by matthewsiemens** [comment]
Hey folks!
I'm trying to do something very similar to what mstn was doing but I'm having a bit of trouble wrapping my head around it. Any chance either of you would be willing to provide some code examples of what this cellFormatter would look like and where to add it?

Thanks so much!

**Comment by mstn** [comment]
@matthewsiemens here you go some pseudocode. I didn't check it, but you should be able to find out the correct syntax easily.

```javascript
const doEdit = (itemId) => {
   // do something
}

const renderEditAction = (action) => {
   return <IconButton name="edit" icon="edit" onClick={()=>doEdit(action.itemId)}/>
}

<DataTable rows={[
   { id: '1', description:'This is a description', action:{ type: 'edit', itemId: '1'} }
]}>
   <TableHeader name='description'>Description</TableHeader>
   <TableHeader name='action' cellFormatter={renderEditAction}>Edit</TableHeader>
</DataTable>
```

You can have also multiple actions per row. What I did in my project is to wrap the original DataTable inside a custom one that adds n actions to each row.

**Comment by matthewsiemens** [comment]
Hey Marco!

That's awesome, thanks so much for taking the time to respond!








*--Matthew SiemensWebsite:http://matthewsiemens.com
<http://siemenstech.ca/>Cell:(306) 202-6574*
